lib/Drush/Command/InitCommand.php

```diff
@@ -0,0 +1,174 @@
+<?php
+namespace Drush\Command;
+
+/**
+ * @file
+ *   Set up local Drush configuration.
+ */
+
+use Drush\Log\LogLevel;
+
+class InitCommand extends \Robo\Tasks
```

Comment:
Commandfile discovery hasn't been designed yet, exactly. The basic mechanism is similar to what Drush does today: find all commandfiles, and then register all of the commands they define. Note that annotations on the class, or using instanceof are both possibilities for secondary techniques for command discovery / selection; however, there needs to be an additional initial filter function as well, as we would not want to load every class file available just to determine which are commandfiles.

DrupalConsole searches for files called *Command.php (in certain paths), and registers all of those. We should probably do something similar. Suggestions / PRs welcome.